### DISCUSSION TITLE
How to commit each message in withHandler

### DISCUSSION THREAD

**Discussion by lemon777** [mateusjunges/laravel-kafka#306] [Q&A]
Please show me the way how to commit message in withHandler
```
$consumer = Kafka::consumer()
            ->subscribe('topic')
            ->enableBatching()
            ->withBatchSizeLimit(1000)
            ->withBatchReleaseInterval(1500)
            ->withHandler(function (\Illuminate\Support\Collection $collection, \Junges\Kafka\Contracts\MessageConsumer $consumer) {
                /* How to commit message here */
            })
            ->build();

$consumer->consume();
```

**Comment by mateusjunges** [comment]
Hey @lemon777 👋 
You don't commit messages in the message handler. By default, the `DefaultCommitterFactory` is used to create a `BatchCommitter`, when auto committing, or a `VoidCommitter`, when auto commit is disabled. You can create your own committer factory by creating a class that implements the `CommitterFactory` contract. This interface gives you the following method:

```php
public function make(KafkaConsumer $kafkaConsumer, Config $config): CommitterContract
{
    //
}
```

This factory must return an instance of the `Committer` contract. Then, you can use the `usingCommitterFactory` method to configure your consumer to use the new committer. You should be able to add your own logic there for how to commit messages.

**Reply by talenta-eg** [reply]
**You should be able to add your own logic there for how to commit messages**

Please could you provide us a reference to implement commit messages manually  ,after check if  process message is done
